### ISSUE TITLE
The result of Clothing1M can't be reproduced

### ISSUE THREAD

**Issue by zysxmu** [microsoft/MLC#6]
@apsyx 
Dear author, we have tried to reproduce the result of your method on Clothing1M.
We use the file in #4 to generate the clean_train, clean_val, noisy_train, clean_test folders.
But the result we obtained is still lower than the result in the paper.

**Comment by zysxmu** [comment]
The result is:
![image](https://user-images.githubusercontent.com/95413091/147866534-3b476288-1cc3-43ac-8685-7534be128ffd.png)

**Comment by randydkx** [comment]
@zysxmu Hi, can you reproduce results in other settings of the paper, I found that it's really time consuming to conduct the experiments and I think the meta label correction module is hard to converge.

**Comment by apsyx** [comment]
Hi @zysxmu and @randydkx, thank you for your interest in MLC. 

For clothing1m with real-world noisy labels (which tends to be not that noisy), what we found earlier is that it's easier for MLC to learn by formulating the meta-model as following:

**y_c = alpha * y' + (1-alpha) * g(h(x), y')**

where alpha is a learnable mixing parameter (also part of the entire meta-model). This effectively adds a skip link from the noisy label to the corrected label with learnable weights.

This skip link formulation was not discussed in the AAAI paper due to space limitation (it's originally briefly discussed in our arxiv preprint v1 - Sec 3.1 in [here](https://arxiv.org/pdf/1911.03809v1.pdf)). We've updated the code for this; you can check the updated `meta_model.py` to see how this is implemented. And this skip link behavior can be turned on by using the `--skip` switch in the command.

Regarding experiments on cloghting1m, you can run the command with 

`python3 main.py --dataset clothing1m --optimizer sgd --bs 32 --corruption_type unif --corruption_level 0.1 --gold_fraction 0.1 --epochs 15 --main_lr 0.1 --meta_lr 0.003 --runid clothing1m_run --cls_dim 128 --skip --gradient_steps 5`

On my current environment setup (pytorch 1.10 with cuda11 on an RTX A6000), this gives around 75% test accuracy. The project README is also updated to reflect this.

**Comment by apsyx** [comment]
P.S. A general note for tuning MLC: Like many other bi-level optimization frameworks, MLC is quite sensitive to the choice of hyper-parameters and meta-model architecture specification. A not well-tuned pair of learning rates (main_lr and meta_lr) is likely to cause the framework to underperform. While we didn't extensively explore the best meta-model architecture, we find it beneficial to tune (in decreasing order of importance): both learning rates, skip switch on/off and number of inner loop gradient steps for your problem setting. Hope this helps!

**Comment by sangkeun00** [comment]
@apsyx 
Thanks for very interesting work! I read your paper where you wrote that you only used 1k clean examples for meta-training for the fair comparison. However, I don't see the current code enforcing this data sampling restriction. If you could provide clarification on this, that would be much appreciated!

Best,
Sang
